lib/composer_element_template.ex

```diff
@@ -0,0 +1,127 @@
+              ]
+  '''
+
+  @impl true
+  def handle_init(_options) do
+    state = %{frames_queues: %{first_queue: :queue.new(), second_queue: :queue.new()}}
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_process(pad, buffer, _contex, state) do
+    state = cond do  # updates state with new queue
+      pad == :first_input -> # it's boilerplate, but it works
+        updated_first_queue = state.frames_queues.first_queue
+        updated_first_queue = :queue.in(buffer, updated_first_queue)
+        updated_frames_queues = state.frames_queues
+        updated_frames_queues = %{updated_frames_queues | first_queue: updated_first_queue}
+        %{state | frames_queues: updated_frames_queues}
+      pad == :second_input ->
+        updated_second_queue = state.frames_queues.second_queue
+        updated_second_queue = :queue.in(buffer, updated_second_queue)
+        updated_frames_queues = state.frames_queues
+        updated_frames_queues = %{updated_frames_queues | second_queue: updated_second_queue}
+        %{state | frames_queues: updated_frames_queues}
+    end
+
+    case {:queue.out(state.frames_queues.first_queue), :queue.out(state.frames_queues.second_queue)} do
+      {{{:value, first_frame_buffer}, rest_of_first_queue}, {{:value, second_frame_buffer}, rest_of_second_queue}} ->  # checks if both queues are not empty
+        {:ok, merged_frame_binary} = merge_frames(first_frame_buffer.payload, second_frame_buffer.payload)
+        merged_image_buffer = %Buffer{first_frame_buffer | payload: merged_frame_binary}  # packs merged binary into buffer (for encoder)
+        frames_queues = %{first_queue: rest_of_first_queue, second_queue: rest_of_second_queue}  # updates queues (removes processed frames)
+        state = %{state | frames_queues: frames_queues}  # updates state with reduced queues
+        {{:ok, buffer: {:output, merged_image_buffer}}, state}  # sends buffer to encoder
+      _ ->  # one of queues is empty
+        {:ok, state}
+    end
+  end
+
+  @impl true
+  def handle_caps(:first_input, %RawVideo{width: 1280, height: 720} = caps, _context, state) do
+    caps = %{caps | height: caps.height * 2}
+    {{:ok, caps: {:output, caps}}, state}
+  end
+
+  @impl true
+  def handle_caps(:second_input, %RawVideo{width: 1280, height: 720} = caps, _context, state) do
+    caps = %{caps | height: caps.height * 2}
+    {{:ok, caps: {:output, caps}}, state}
+  end
+
+
+  @impl true
+  def handle_end_of_stream(pad, _context, state) do
+    state = cond do
+      pad == :first_input ->
+        updated_frames_queues = state.frames_queues
+        updated_frames_queues = %{updated_frames_queues | first_queue: :end_of_the_stream}
+        %{state | frames_queues: updated_frames_queues}
+      pad == :second_input ->
+        updated_frames_queues = state.frames_queues
+        updated_frames_queues = %{updated_frames_queues | second_queue: :end_of_the_stream}
+        %{state | frames_queues: updated_frames_queues}
+    end
```

Comment:
What if the queue is not empty? You'll skip some buffers in that case. Maybe just put eos at the end of queue (and replace the queue with eos if it's a last element) - or keep a separate list of an elements with eos.